### ISSUE TITLE
referencePaths

### ISSUE THREAD

**Issue by yuyaryshev** [babel/babel#10299]
## Bug Report

**Current Behavior**
An identifier is not counted in referencePaths for variable assingnment.
Reference from line 2 in example below is missing.

**Input Code**
```js
        let a = 1;
        a = 2;
        console.log('a=', a);
```

**Expected behavior/code**
referencePaths for binding "a" to contain 3 references.

**Environment**
- Babel version(s): 7.5.5

**Comment by nicolo-ribaudo** [comment]
I would expect only the second like and the third line to count as references; the fiesta line is a binding.

**Comment by JLHwung** [comment]
True, and it seems that https://github.com/babel/babel/blob/5fb4d84a33351c13057dc542513a3fe2309e08b0/packages/babel-types/src/validators/isReferenced.js#L77-L81

does not conform to [§6.2.4](https://tc39.es/ecma262/#sec-reference-specification-type)
> the left-hand operand of an assignment is expected to produce a reference.
